pwndbg/commands/dt.py

```diff
@@ -16,20 +14,25 @@
     Optionally overlay that information at an address.
     """,
 )
-parser.add_argument("typename", type=str, help="The name of the structure being dumped.")
+parser.add_argument(
+    "typename",
+    type=str,
+    help='The name of the structure being dumped. Use quotes if the type contains spaces (e.g. "struct malloc_state").',
+)
 parser.add_argument(
     "address", type=int, nargs="?", default=None, help="The address of the structure."
 )
 
 
 @pwndbg.commands.ArgparsedCommand(parser)
-def dt(typename: str, address: int | pwndbg.dbg_mod.Value | None = None) -> None:
+def dt(typename: str, address: int | None = None) -> None:
     """
     Dump out information on a type (e.g. ucontext_t).
 
     Optionally overlay that information at an address.
     """
-    if address is not None:
-        address = pwndbg.commands.fix(str(address))
+    if address is not None and not pwndbg.aglib.memory.is_readable_address(address):
+        print(message.error("The provided address is not readable."))
```

Comment:
This is just a comment, not for this PR I think.
we could fix this globally with new type parser? eg: `type=parse_as_valid_address`
and remove weird behavior with `type=int` (fix_int) (because under the hood we are overriding it with gdb.parse_and_eval... 🤮 )